### DISCUSSION TITLE
Does MediaMTX support adaptive bitrate streaming?

### DISCUSSION THREAD

**Discussion by CheSauRie** [bluenviron/mediamtx#3953] [Questions]
### Question

Hello everyone,

I am currently using MediaMTX and would like to know if it supports adaptive bitrate streaming (ABR). If it does, could you please provide some guidance or documentation on how to configure it?

**Comment by JoshuaHintze** [comment]
I would like to know if anyone has this working too.

**Comment by ljwall** [comment]
I could be wrong about this, but as far as I can see mediamtx does not do any video decoding / re-encoding, it is repackaging the encoded video to different container formats and transport protocols, but performs no re-encoding.   If that's right I don't see how it could provide any kind of ABR to clients out of the box.

**Reply by MatisOne** [reply]
You can now stream [multiple video tracks (and multiple audio tracks)](https://github.com/bluenviron/mediamtx/pull/4088) using Enhanced RTMP(which is fully released in version [v1.11.0](https://github.com/bluenviron/mediamtx/releases/tag/v1.11.0)). However, there was a way to do this using [this](https://github.com/bluenviron/mediamtx?tab=readme-ov-file#remuxing-re-encoding-compression) even before E-RTMP was a thing (by ffmpeg).
